### PR TITLE
Errkit migration 2 (pkg/kopia)

### PR DESCRIPTION
**Pull request by e-sumin**
## Change Overview

In accordance with decision to migrate to our own errors package (look here https://github.com/kanisterio/kanister/issues/1838), this PR contains migration of pkg/kopia to errkit package.
Further migration PRs are expected.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kopia/repository/client.go [hunk 1]
```diff
@@ -92,7 +91,7 @@ func ConnectToAPIServer(
 		}
 		return true, nil
 	})
-	return errors.Wrap(err, "Failed connecting to the Kopia API Server")
+	return errkit.Wrap(err, "Failed connecting to the Kopia API Server")
```
Comment (hairyhum) [inline pkg/kopia/repository/client.go:94]: Check for `err != nil`

File: pkg/kopia/repository/client.go [hunk 2]
```diff
@@ -101,24 +100,24 @@ func ConnectToAPIServer(
...
+		return nil, errkit.Wrap(err, "Failed to open kopia repository")
 	}
 
 	_, rw, err := r.NewWriter(ctx, repo.WriteSessionOptions{
 		Purpose:  purpose,
 		OnUpload: func(i int64) {},
 	})
 
-	return rw, errors.Wrap(err, "Failed to open kopia repository writer")
+	return rw, errkit.Wrap(err, "Failed to open kopia repository writer")
```
Comment (hairyhum) [inline pkg/kopia/repository/client.go:120]: Check for `err != nil`

File: pkg/kopia/repository/connect_or_create.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func ConnectToOrCreateKopiaRepository(
 	case err == nil:
 		// If repository connect was successful, we're done!
 		return nil
-	case kerrors.IsInvalidPasswordError(err):
+	case kerrors.IsInvalidPasswordError(err): // TODO: Verify
```
Comment (hairyhum) [inline pkg/kopia/repository/connect_or_create.go:44]: Verify what?

File: pkg/kopia/snapshot/stream.go [hunk 4]
```diff
@@ -198,31 +197,31 @@ func Read(ctx context.Context, target io.Writer, backupID, path, password string
...
-		return errors.Wrapf(err, "Failed to open kopia object: %v", oid)
+		return errkit.Wrap(err, "Failed to open kopia object", "oid", oid)
 	}
 
 	defer r.Close() //nolint:errcheck
 
 	_, err = copy(target, r)
 
-	return errors.Wrap(err, "Failed to copy snapshot data to the target")
+	return errkit.Wrap(err, "Failed to copy snapshot data to the target")
```
Comment (hairyhum) [inline pkg/kopia/snapshot/stream.go:207]: Check `err != nil`?

File: pkg/kopia/snapshot/stream.go [hunk 5]
```diff
@@ -236,7 +235,7 @@ func ReadFile(ctx context.Context, backupID, target, password string) error {
 	_, err = restore.Entry(ctx, rep, output, rootEntry, restore.Options{
 		Parallel: 8,
 	})
-	return errors.Wrap(err, "Failed to copy snapshot data to the target")
+	return errkit.Wrap(err, "Failed to copy snapshot data to the target")
```
Comment (hairyhum) [inline pkg/kopia/snapshot/stream.go:238]: Check err != nil?

**Review by hairyhum — COMMENTED** [review]
Maybe we should return `nil` when wrapping `nil` errors?

It's really easy to miss, it's missed several times in this PR. 

Especially since we have things like pkg/kopia/repository/connect.go:31

File: pkg/kopia/repository/client.go (hunk 1 above)
Comment (e-sumin) [reply pkg/kopia/repository/client.go]: Done

File: pkg/kopia/repository/client.go (hunk 2 above)
Comment (e-sumin) [reply pkg/kopia/repository/client.go]: Done

File: pkg/kopia/repository/connect_or_create.go (hunk 3 above)
Comment (e-sumin) [reply pkg/kopia/repository/connect_or_create.go]: Verify that matching is not broken. I've added test for error matchers.

File: pkg/kopia/snapshot/stream.go (hunk 4 above)
Comment (e-sumin) [reply pkg/kopia/snapshot/stream.go]: Done,

File: pkg/kopia/snapshot/stream.go (hunk 5 above)
Comment (e-sumin) [reply pkg/kopia/snapshot/stream.go]: Done

**Review by hairyhum — APPROVED** [review]

File: pkg/kopia/command/parse_command_output.go [hunk 6]
```diff
@@ -23,13 +23,12 @@ import (
 	"strings"
 
 	"github.com/dustin/go-humanize"
+	"github.com/kanisterio/errkit"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/log"
```
Comment (PrasadG193) [inline pkg/kopia/command/parse_command_output.go:28]: Let's put internal packages in a separate group

File: pkg/kopia/repository/client.go [hunk 7]
```diff
@@ -22,13 +22,12 @@ import (
...
-	"github.com/kopia/kopia/repo"
-	"github.com/kopia/kopia/repo/content"
-	"github.com/pkg/errors"
-
+	"github.com/kanisterio/errkit"
 	"github.com/kanisterio/kanister/pkg/kopia"
 	"github.com/kanisterio/kanister/pkg/log"
 	"github.com/kanisterio/kanister/pkg/poll"
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/content"
```
Comment (PrasadG193) [inline pkg/kopia/repository/client.go:30]: Please keep kanister and other packages into separate groups

File: pkg/kopia/repository/connect_or_create.go [hunk 8]
```diff
@@ -19,6 +19,7 @@ import (
 
 	"k8s.io/client-go/kubernetes"
 
+	"github.com/kanisterio/errkit"
 	"github.com/kanisterio/kanister/pkg/kopia/command"
```
Comment (PrasadG193) [inline pkg/kopia/repository/connect_or_create.go:23]:
```suggestion
	"k8s.io/client-go/kubernetes"
	"github.com/kanisterio/errkit"
	
	"github.com/kanisterio/kanister/pkg/kopia/command"
```
errkit package should be considered as external package and put it into separate group

File: pkg/kopia/snapshot/snapshot.go [hunk 9]
```diff
@@ -21,16 +21,15 @@ import (
...
 	"github.com/kopia/kopia/fs"
 	"github.com/kopia/kopia/repo"
 	"github.com/kopia/kopia/repo/manifest"
 	"github.com/kopia/kopia/snapshot"
 	"github.com/kopia/kopia/snapshot/policy"
 	"github.com/kopia/kopia/snapshot/snapshotfs"
-	"github.com/pkg/errors"
-
-	"github.com/kanisterio/kanister/pkg/kopia"
-	"github.com/kanisterio/kanister/pkg/kopia/repository"
```
Comment (PrasadG193) [inline pkg/kopia/snapshot/snapshot.go:33]: `github.com/kanisterio/kanister` are internal packages, so should be kept in a separate group

File: pkg/kopia/snapshot/stream.go [hunk 10]
```diff
@@ -21,16 +21,15 @@ import (
 	"path/filepath"
 	"sync"
 
+	"github.com/kanisterio/errkit"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
```
Comment (PrasadG193) [inline pkg/kopia/snapshot/stream.go:26]: same here

**Review by PrasadG193 — APPROVED** [review]
Need to correct import grouping. Rest LGTM

**Comment by e-sumin** [comment]
@PrasadG193 
```
goimports -w -d -local github.com/kanisterio/kanister
```
has been applied, so now our packages are properly sorted

**Pull request merged**